Gems/ROS2/Code/Source/Frame/ROS2FrameComponent.cpp

```diff
@@ -65,9 +78,27 @@ namespace ROS2
 
         if (m_publishTransform)
         {
+            AZ_TracePrintf("ROS2FrameComponent", "Setting up %s", GetFrameID().data());
+
+            // The frame is static when is not a top entity and one of the following is true:
+            //    - if the entity has at least one Fixed Joint
+            //    - if the entity has no joints at all.
```

Comment:
nit: this is super minor, but this comment is describing the inversed condition of what the code is doing, making it hard to follow. I'd recommend describing the the positive condition rather than the inversed. Example:

````
            // The frame will always be dynamic if it's a top entity.
            if (IsTopLevel())
            {
                m_isDynamic = true;
            }
            // Otherwise it'll be dynamic when it has joints and and it's not a fixed joint.
            else
            {
                bool hasJoints = Internal::CheckIfEntityHasComponentOfType(
                    m_entity, AZ::Uuid("{B01FD1D2-1D91-438D-874A-BF5EB7E919A8}")); // Physx::JointComponent;
                bool hasFixedJoints = Internal::CheckIfEntityHasComponentOfType(
                    m_entity, AZ::Uuid("{02E6C633-8F44-4CEE-AE94-DCB06DE36422}")); // Physx::FixedJointComponent
                m_isDynamic = hasJoints && !hasFixedJoints;
            }
````

Reply:
Looks great now! thank you for your suggestion!